tools/serve/serve.py

```diff
@@ -360,6 +360,7 @@ class ShadowRealmHandler(HtmlWrapperHandler):
   await new Promise(r.evaluate(`
     (resolve, reject) => {
       (async () => {
+        globalThis.TESTHARNESSJS_IS_INSIDE_SHADOW_REALM = true;
```

Comment:
In other autogenerated tests in this file, we use
```
self.GLOBAL = {
  isWindow: function() { return true; },
  isWorker: function() { return false; },
};
```
Maybe use something like that?

Reply:
I wasn't sure if it is important that those are set _after_ loading testharness. I'll add it to that code instead. This should really be a `self.GLOBAL.env` IMO, with values `"window" | "worker" | "shadowrealm"`. But whatever, that's something for another day :-)

Reply:
Actually can't use `self.GLOBAL`, because it isn't set in all tests. I think the previous iteration where it was a separate testharness.js specific global is better. Also `self.GLOBAL` is not something testharness.js relies on right now.

We can still add `GLOBAL.isShadowRealm` but that seems like something for a separate PR.